### PR TITLE
A few improvements of the script to create new Boskos projects

### PR DESCRIPTION
**Pull request by chizhg**
**What this PR does, why we need it**:
As a followup of https://github.com/knative/test-infra/pull/1363

/cc @adrcunha

### CODE REVIEW

File: ci/prow/boskos/create_projects.sh [hunk 1]
```diff
@@ -16,18 +16,29 @@
...
-readonly NUMBER=${2:?"Second argument is the number of new projects."}
-readonly BILLING_ACCOUNT=${3:?"Third argument must be the billing account."}
-readonly OUTPUT_FILE=${4:?"Fourth argument should be a file name all project names will be appended to in a resources.yaml format."}
+readonly NUMBER=${1:?"First argument is the number of new projects to create."}
+readonly BILLING_ACCOUNT=${2:?"Second argument must be the billing account."}
 
+readonly RESOURCE_FILE="resources.yaml"
+
+if [[ ! -f ${RESOURCE_FILE} || ! -w ${RESOURCE_FILE} ]]; then
+  echo "${RESOURCE_FILE} does not exist or is not writable"
```
Comment (adrcunha) [inline ci/prow/boskos/create_projects.sh]: Exit here.

File: ci/prow/boskos/create_projects.sh [hunk 2]
```diff
@@ -16,18 +16,29 @@
...
+fi
+
+# Get the index of the last boskos project from the resources file
+START=$(grep -n "knative-boskos-" ${RESOURCE_FILE} | wc -l)
+((START++))
```
Comment (adrcunha) [inline ci/prow/boskos/create_projects.sh]: Rename START to LAST_INDEX, no need to increment here, start loop below from 1.

File: ci/prow/boskos/create_projects.sh [hunk 3]
```diff
@@ -16,18 +16,29 @@
...
   # If this needs to be changed for any reason, GCP project settings must be updated.
   # Details are available in Google's internal issue 137963841.
   gcloud projects create ${PROJECT} --folder=396521612403
   gcloud beta billing projects link ${PROJECT} --billing-account=${BILLING_ACCOUNT}
+
+  # Set permissions for this project
   "$(dirname $0)/set_permissions.sh" ${PROJECT}
-  echo "  - ${PROJECT}" >> ${OUTPUT_FILE}
+
+  last_project=$(grep "knative-boskos-" resources.yaml | tail -1)
```
Comment (adrcunha) [inline ci/prow/boskos/create_projects.sh]: not: LAST_PROJECT

File: ci/prow/boskos/create_projects.sh (hunk 1 above)
Comment (chizhg) [reply ci/prow/boskos/create_projects.sh]: Done.

File: ci/prow/boskos/create_projects.sh (hunk 3 above)
Comment (chizhg) [reply ci/prow/boskos/create_projects.sh]: Done.

File: ci/prow/boskos/create_projects.sh (hunk 2 above)
Comment (chizhg) [reply ci/prow/boskos/create_projects.sh]: Done.

File: ci/prow/boskos/create_projects.sh [hunk 4]
```diff
@@ -22,14 +22,13 @@ readonly BILLING_ACCOUNT=${2:?"Second argument must be the billing account."}
 readonly RESOURCE_FILE="resources.yaml"
 
 if [[ ! -f ${RESOURCE_FILE} || ! -w ${RESOURCE_FILE} ]]; then
-  echo "${RESOURCE_FILE} does not exist or is not writable"
+  echo "${RESOURCE_FILE} does not exist or is not writable" && exit 1
```
Comment (adrcunha) [inline ci/prow/boskos/create_projects.sh]: There's no reason to make this a single line, hurting readability. Please split.

Comment (chizhg) [reply ci/prow/boskos/create_projects.sh]: Done.

File: ci/prow/boskos/create_projects.sh [hunk 5]
```diff
@@ -16,18 +16,29 @@
...
-  PROJECT="knative-boskos-$(( i + ${FIRST} ))"
+readonly RESOURCE_FILE="resources.yaml"
+
+if [[ ! -f ${RESOURCE_FILE} || ! -w ${RESOURCE_FILE} ]]; then
+  echo "${RESOURCE_FILE} does not exist or is not writable"
+  exit 1
+fi
+
+# Get the index of the last boskos project from the resources file
+LAST_INDEX=$(grep -n "knative-boskos-" ${RESOURCE_FILE} | wc -l)
```
Comment (adrcunha) [inline ci/prow/boskos/create_projects.sh]: It's safer not relying on `wc`:

```suggestion
LAST_INDEX=$(grep "knative-boskos-" ${RESOURCE_FILE} | grep -o "[0-9]\+" | sort -nr | head -1)
```

File: ci/prow/boskos/create_projects.sh [hunk 6]
```diff
@@ -16,18 +16,29 @@
...
   # If this needs to be changed for any reason, GCP project settings must be updated.
   # Details are available in Google's internal issue 137963841.
   gcloud projects create ${PROJECT} --folder=396521612403
   gcloud beta billing projects link ${PROJECT} --billing-account=${BILLING_ACCOUNT}
+
+  # Set permissions for this project
   "$(dirname $0)/set_permissions.sh" ${PROJECT}
-  echo "  - ${PROJECT}" >> ${OUTPUT_FILE}
+
+  LAST_PROJECT=$(grep "knative-boskos-" resources.yaml | tail -1)
```
Comment (adrcunha) [inline ci/prow/boskos/create_projects.sh]:
```suggestion
  LAST_PROJECT=$(grep "knative-boskos-" ${RESOURCE_FILE} | tail -1)
```

File: ci/prow/boskos/create_projects.sh (hunk 5 above)
Comment (chizhg) [reply ci/prow/boskos/create_projects.sh]: Done.

File: ci/prow/boskos/create_projects.sh (hunk 6 above)
Comment (chizhg) [reply ci/prow/boskos/create_projects.sh]: Done.

**Comment by adrcunha** [comment]
More automation, less manual stuff. :)

/approve
/lgtm

**Pull request merged**